### ISSUE TITLE
Implement DatasetMerger (multi-dataset weighted sampling)

### ISSUE THREAD

**Issue by timlawrenz** [timlawrenz/DINO-X#11]
## Summary
Combine multiple dataset manifests into a single training set with configurable sampling strategies.

## Features
- **Weighted sampling** — prevent large datasets from dominating (e.g., TotalSegmentator 1.5M vs LIDC 235K)
- **Modality filtering** — select only CT, only MRI, etc.
- **Organ filtering** — pan-organ or organ-specific subsets
- **Spacing-stratified sampling** — ensure diverse spacing coverage for scale embedding training
- **Deterministic** — given same seed, produces same training order

## API
```python
merger = DatasetMerger()
merger.add("lidc-idri", weight=0.40)
merger.add("ct-org", weight=0.25)
merger.add("abdomenct-1k", weight=0.35)
merged = merger.build(seed=42)
# Returns unified manifest with source tracking
```

## Depends On
- #9 (DataManifest)

**Comment by timlawrenz** [comment]
## Warning: DINO Oversampling Collapse Risk

The `DatasetMerger` is implemented and working (migrated to `zoo/merge.py`), but there is a critical consideration for self-supervised training that must be documented.

### The Problem

When mixing heavily imbalanced datasets — e.g., LIDC-IDRI (235K slices) + a smaller dataset (20K slices) — naively up-weighting the small dataset to achieve balanced batches means **the same 20K slices appear ~12× each per epoch**.

In supervised learning, this is standard minority oversampling. In **DINO self-supervised learning**, this is dangerous:

- The student-teacher loss minimizes distance between embeddings of augmented views
- If the model sees the same slices repeatedly, it can minimize loss by **memorizing** those specific images rather than learning generalizable features
- This drives **representation collapse** — embeddings for oversampled slices cluster tightly, losing discriminative power

### Current Mitigations (already in place)

Our augmentation pipeline provides significant view diversity per slice:
1. **Random windowing**: Level ∈ [-400, 400], Width ∈ [800, 2000] — each slice gets different tissue contrast
2. **RandomResizedCrop**: scale (0.5–1.0) — different spatial regions each time
3. **RandomHorizontalFlip**: 50% probability
4. **2.5D stacking**: 3 consecutive slices → pseudo-RGB, so the model sees volumetric context

The combination of random windowing + random crop means the model almost never sees the exact same pixel values twice, even for the same slice. This is a strong defense against memorization.

### Recommendation for `DatasetMerger.build()`

For the MVP (24K LIDC + 18K Pancreas ≈ 1.4:1 ratio), the imbalance is mild. But for future pan-organ models mixing 1M+ slice datasets with 20K niche datasets:

1. **Cap oversampling ratio** — Never repeat a slice more than 5× per epoch. If weights demand more, clip and log a warning.
2. **Monitor per-dataset embedding variance** — If σ for the oversampled dataset drops below σ for the large dataset, collapse is occurring.
3. **Consider epoch-level weighting** — Instead of sampling N×weight slices per dataset, use all slices from each dataset but weight the loss contribution per batch by dataset origin. This preserves view diversity while still balancing gradient influence.

The current MVP fleet run uses a ~58/42 split (24K/18K), which is within safe bounds.

**Comment by timlawrenz** [comment]
✅ Implemented in `zoo/merge.py`: DatasetMerger with weighted sampling and configurable oversampling cap.

Note: DINO oversampling collapse risk documented — cap at 5× to prevent representation collapse from repeated identical slices (see issue comments for analysis).
Commit: 432b5c5
